Deploy/stacks/dynamic/stack-clients/src/main/java/com/cmclinnovations/stack/clients/postgis/PostGISClient.java

```diff
@@ -77,4 +108,122 @@ public RemoteRDBStoreClient getRemoteStoreClient(String database) {
+                String osmFileName = f.toString().substring(0, f.toString().lastIndexOf('.'));
+                if (!osmFileName.endsWith(".osm")) {
+                    osmFileName = osmFileName + ".osm";
+                }
+                convertPBF2OSM(f.toString(), osmFileName, getContainerId("postgis"));
+                tempOSMFilesList.add(Path.of(osmFileName));
+            });
+
+            tempOSMFilesList.stream().findFirst().ifPresent(osmFile -> uploadRoutingToPostGIS(database, osmFile,
+                    tempConfigFilePath, tablePrefix, options, append));
+            tempOSMFilesList.stream().skip(1).forEach(osmFile -> uploadRoutingToPostGIS(database, osmFile,
+                    tempConfigFilePath, tablePrefix, options, true));
+        }
+    }
+
+    public void uploadRoutingToPostGIS(String database, Path osmFilePath, Path configFilePath, String tablePrefix,
+            Options options, boolean append) {
+        String containerId = getContainerId("postgis");
+        ensurePostGISRoutingSupportEnabled(database, containerId);
+
+        ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+        ByteArrayOutputStream errorStream = new ByteArrayOutputStream();
+
+        String execId = createComplexCommand(containerId,
+                constructOSM2PGRoutingCommand(osmFilePath.toString(), configFilePath.toString(), database, tablePrefix,
+                        options,
+                        append))
+                .withOutputStream(outputStream)
+                .withErrorStream(errorStream)
+                .withEvaluationTimeout(300)
+                .exec();
+
+        handleErrors(errorStream, execId, logger);
+    }
+
+    private String[] constructOSM2PGRoutingCommand(String osmFile, String configFile, String database,
+            String tablePrefix, Options options,
+            boolean append) {
+        List<String> command = new ArrayList<>(Arrays.asList("osm2pgrouting", "--f", osmFile, "--conf",
+                configFile, "--dbname", database, "--username", postgreSQLEndpoint.getUsername(),
+                "--password", postgreSQLEndpoint.getPassword(), "--prefix", tablePrefix));
+        if (!append) {
+            command.add("--clean");
+        }
+        command.addAll(options.getOptionsList());
+
+        return command.toArray(String[]::new);
+    }
+
+    private void ensurePostGISRoutingSupportEnabled(String database, String postGISContainerId) {
+        ByteArrayOutputStream errorStream = new ByteArrayOutputStream();
+        String execId = createComplexCommand(postGISContainerId,
+                "psql", "-U", postgreSQLEndpoint.getUsername(), "-d", database, "-w")
+                .withHereDocument(
+                        "CREATE EXTENSION IF NOT EXISTS pgrouting CASCADE; CREATE EXTENSION IF NOT EXISTS hstore;")
+                .withErrorStream(errorStream)
+                .exec();
+        handleErrors(errorStream, execId, logger);
+    }
```

Comment:
Might it make sense to move this to a new class, for example ```PGRoutingClient```?

Reply:
Resolved by 8060bd9